src/Pact/Types/Runtime.hs

```diff
@@ -213,6 +219,17 @@ makeLenses ''RefState
 instance NFData RefState
 instance Default RefState where def = RefState HM.empty HM.empty Nothing
 
+data PactEvent = PactEvent
+  { _eventName :: QualifiedName
+  , _eventParams :: [PactValue]
+  , _eventModuleHash :: ModuleHash
+  } deriving (Eq, Show, Generic)
```

Comment:
Strictify?

Reply:
Yes, and I also broke into `Text` + `ModuleName`, mainly because of the `Info` in qualified name messing up the JSON, but also allows for indexing on name/topic alone (ie all `TRANSFER`s, not just `coin.TRANSFER` etc)